tough/src/lib.rs

```diff
@@ -328,14 +334,21 @@ impl<'a, T: Transport> Repository<'a, T> {
         //   HASH is one of the hashes of the targets file listed in the targets metadata file
         //   found earlier in step 4. In either case, the client MUST write the file to
         //   non-volatile storage as FILENAME.EXT.
-        Ok(
-            if let Some(target) = self.targets.signed.targets.get(name) {
-                let (sha256, file) = self.target_digest_and_filename(target, name);
-                Some(self.fetch_target(target, &sha256, file.as_str())?)
-            } else {
-                None
-            },
-        )
+        Ok(if let Ok(target) = self.targets.signed.find_target(name) {
+            let (sha256, file) = self.target_digest_and_filename(target, name);
+            Some(self.fetch_target(target, &sha256, file.as_str())?)
+        } else {
+            None
+        })
+    }
+
+    pub fn get_role(&self, name: &str) -> Result<&DelegatedRole> {
```

Comment:
Since root.json, targets.json, etc are all "Roles", maybe this should be called `pub fn delegate(&self, name: &str)`? And the error `DelegateNotFound`?

Reply:
I don't know if delegate is the term we want to use for the roles that have been delegated as they are still roles and with 2 meanings of the word delegate the naming could create confusion.